homeassistant/components/google_assistant/trait.py

```diff
@@ -609,6 +612,58 @@ async def execute(self, command, data, params, challenge):
         )
 
 
+class EnergyStorageTrait(_Trait):
+    """Trait to offer EnergyStorage functionality.
+
+    https://developers.google.com/actions/smarthome/traits/energystorage
+    """
+
+    name = TRAIT_ENERGYSTORAGE
+    commands = [COMMAND_CHARGE]
+
+    @staticmethod
+    def supported(domain, features, device_class, _):
+        """Test if state is supported."""
+        return domain == vacuum.DOMAIN and features & vacuum.SUPPORT_BATTERY
+
+    def sync_attributes(self):
+        """Return dock attributes for a sync request."""
+        return {
+            "isRechargeable": True,
+            "queryOnlyEnergyStorage": True,
+        }
+
+    def query_attributes(self):
+        """Return dock query attributes."""
+        battery_level = self.state.attributes.get(ATTR_BATTERY_LEVEL)
+        if battery_level == 100:
+            descriptive_capacity_remaining = "FULL"
+        elif battery_level >= 75 and battery_level < 100:
+            descriptive_capacity_remaining = "HIGH"
+        elif battery_level >= 50 and battery_level < 75:
+            descriptive_capacity_remaining = "MEDIUM"
+        elif battery_level >= 25 and battery_level < 50:
+            descriptive_capacity_remaining = "LOW"
+        elif battery_level >= 0 and battery_level < 25:
+            descriptive_capacity_remaining = "CRITICALLY_LOW"
```

Comment:
Could really use some input here - is there anywhere else in the code where we do a similar mapping, and does it make sense to keep the percentage -> description mapping consistent?

Reply:
I think this is fine for now.